### PR TITLE
Update Boost.java

### PR DESCRIPTION
**Pull request by Niix-Dan**
image

### CODE REVIEW

**Comment by RacoonDog** [comment]
NPE when activated out of a world

**Comment by Niix-Dan** [comment]
> NPE when activated out of a world

NPE? what is this

**Comment by arlomcwalter** [comment]
null pointer exception

**Comment by arlomcwalter** [comment]
you need to add a Utils.canUpdate() check at the start of that method

**Comment by arlomcwalter** [comment]
the boost player method
also idk why you start the name with _ and also fields should go above the constructor

File: src/main/java/anticope/rejects/modules/Boost.java [hunk 1]
```diff
@@ -15,24 +15,57 @@ public class Boost extends Module {
...
     );
+    
+    private final Setting<Boolean> autoBoost = sgGeneral.add(new BoolSetting.Builder()
+        .name("auto-boost")
+        .description("Automatically boosts you.")
+        .defaultValue(false)
+        .build()
+    );
+    
+    private final Setting<Integer> delay = sgGeneral.add(new IntSetting.Builder()
```
Comment (C10udburst) [inline src/main/java/anticope/rejects/modules/Boost.java:31]: Could you change `delay -> delaySetting` and `_delay -> delay`? Its more consistent with java style guide.

```suggestion
    private final Setting<Integer> delaySetting = sgGeneral.add(new IntSetting.Builder()
```

File: src/main/java/anticope/rejects/modules/Boost.java [hunk 2]
```diff
@@ -15,24 +15,57 @@ public class Boost extends Module {
...
+        .range(0, 120)
+        .sliderRange(0, 120)
+        .build()
+    );
 
     public Boost() {
         super(MeteorRejectsAddon.CATEGORY, "boost", "Works like a dash move.");
     }
     
+    private int _delay = 0;
```
Comment (C10udburst) [inline src/main/java/anticope/rejects/modules/Boost.java:45]:
```suggestion
    private int delay = 0;
```

File: src/main/java/anticope/rejects/modules/Boost.java [hunk 3]
```diff
@@ -15,24 +15,57 @@ public class Boost extends Module {
...
+    
     @Override
     public void onActivate() {
-        if (mc.player ==  null) {
-            this.toggle();
-            return;
+        boostPlayer();
+        _delay = delay.get();
```
Comment (C10udburst) [inline src/main/java/anticope/rejects/modules/Boost.java:50]:
```suggestion
        delay = delaySetting.get();
```

**Comment by arlomcwalter** [comment]
bro completely ignored all my comments

**Comment by C10udburst** [comment]
> bro completely ignored all my comments

true lol

**Pull request closed without merging**